packages/plugin-core/src/commands/MoveHeader.ts

```diff
@@ -0,0 +1,311 @@
+import {
+  DendronError,
+  NoteProps,
+  NoteUtils,
+  VaultUtils,
+  getSlugger,
+} from "@dendronhq/common-all";
+import {
+  HistoryService,
+  DendronASTDest,
+  DendronASTTypes,
+  AnchorUtils,
+  Processor,
+  RemarkUtils,
+  MDUtilsV5,
+  ProcMode,
+  visit,
+  Heading,
+  Node,
+} from "@dendronhq/engine-server";
+import { LookupControllerV3 } from "../components/lookup/LookupControllerV3";
+import {
+  NoteLookupProvider,
+  NoteLookupProviderSuccessResp,
+} from "../components/lookup/LookupProviderV3";
+import { DENDRON_COMMANDS } from "../constants";
+import { VSCodeUtils } from "../utils";
+import { BasicCommand } from "./base";
+import _ from "lodash";
+import path from "path";
+import { getEngine } from "../workspace";
+import { EngineAPIService } from "../services/EngineAPIService";
+import { delayedUpdateDecorations } from "../features/windowDecorations";
+import { findReferences } from "../utils/md";
+
+type CommandInput =
+  | {
+      dest?: NoteProps;
+    }
+  | undefined;
+type CommandOpts = {
+  origin: NoteProps;
+  nodesToMove: Node[];
+  modifiedOriginTree: Node;
+  originProc: Processor;
+  engine: EngineAPIService;
+} & CommandInput;
+type CommandOutput = {
+  updated: NoteProps[];
+} & CommandOpts;
+
+export class MoveHeaderCommand extends BasicCommand<
+  CommandOpts,
+  CommandOutput
+> {
+  key = DENDRON_COMMANDS.MOVE_HEADER.key;
+
+  headerNotSelectedError = new DendronError({
+    message: "You must first select the header you want to move.",
+  });
+
+  noActiveNoteError = new DendronError({
+    message: "No note open.",
+  });
+
+  getProc = (engine: EngineAPIService, note: NoteProps) => {
+    return MDUtilsV5.procRemarkParse(
+      { mode: ProcMode.FULL },
+      {
+        engine,
+        fname: note.fname,
+        vault: note.vault,
+        dest: DendronASTDest.MD_DENDRON,
+      }
+    );
+  };
+
+  /**
+   * Recursively check if two given node is identical.
```

Comment:
spelling nit: Recursively check if two given nodes are identical.

And should this comment be something like: Recursively check if two given nodes have identical children?

Reply:
you are right.